src/nestegg.c

```diff
@@ -335,6 +340,9 @@ struct saved_state {
 };
 
 struct frame_encryption {
+  uint8_t num_partitions;
+  uint32_t* partition_offsets;
```

Comment:
Space between `uint32_t` and `*`.

Also please move these fields after the `iv` and `length`.  That'll let `num_partitions` and `signal_byte` pack together naturally rather than inducing unnecessary struct padding.